opnode/rollup/sync/start_test.go

```diff
@@ -90,224 +98,213 @@ type syncStartTestCase struct {
 	Name string
 
 	OffsetL2 uint64
-	EngineL1 string
-	EngineL2 string
-	ActualL1 string
+	EngineL1 string // L1 Chain prior to a re-org or other change
+	EngineL2 string // L2 Chain that follows from L1Chain
+	ActualL1 string // L1 Chain after a change may have occurred
 
 	GenesisL1 rune
 	GenesisL2 rune
 
-	ExpectedNextRefL1 rune
-	ExpectedRefL2     rune
+	ExpectedNextRefsL1 string // The L1 extension to follow (i.e. L1 after the L1 parent in the new L2 Head)
+	ExpectedRefL2      rune   // The new L2 tip after a L1 change that may have occured
 
 	ExpectedErr error
 }
 
+func refToRune(r eth.BlockID) rune {
+	return rune(r.Hash.Bytes()[0])
+}
+
 func (c *syncStartTestCase) Run(t *testing.T) {
 	engL1 := chainL1(c.OffsetL2, c.EngineL1)
 	engL2 := chainL2(engL1, c.EngineL2)
 	actL1 := chainL1(0, c.ActualL1)
 
-	msr := &mockSyncReference{
+	msr := &fakeChainSource{
 		L2: engL2,
 		L1: actL1,
 	}
 
 	genesis := &rollup.Genesis{
-		L1: mockID(c.GenesisL1, c.OffsetL2),
-		L2: mockID(c.GenesisL2, 0),
-	}
-
-	expectedNextRefL1Num := ^uint64(0)
-	for i, id := range c.ActualL1 {
-		if id == c.ExpectedNextRefL1 {
-			expectedNextRefL1Num = uint64(i)
-		}
+		L1: fakeID(c.GenesisL1, c.OffsetL2),
+		L2: fakeID(c.GenesisL2, 0),
 	}
-	if c.ExpectedNextRefL1 == 0 {
-		expectedNextRefL1Num = uint64(0)
-	}
-	expectedNextRefL1 := mockID(c.ExpectedNextRefL1, expectedNextRefL1Num)
 
-	expectedRefL2Num := ^uint64(0)
-	for i, id := range c.EngineL2 {
-		if id == c.ExpectedRefL2 {
-			expectedRefL2Num = uint64(i)
-		}
-	}
-	if c.ExpectedRefL2 == 0 {
-		expectedRefL2Num = uint64(0)
-	}
-	expectedRefL2 := mockID(c.ExpectedRefL2, expectedRefL2Num)
+	nextRefL1s, refL2, err := V3FindSyncStart(context.Background(), msr, genesis)
 
-	nextRefL1, refL2, err := FindSyncStart(context.Background(), msr, genesis)
 	if c.ExpectedErr != nil {
-		assert.Error(t, err, "got next L1 %s (%d), onto L2: %s (%d)", nextRefL1.Hash[:1], nextRefL1.Number, refL2.Hash[:1], refL2.Number)
+		assert.Error(t, err, "Expecting an error in this test case")
 		assert.ErrorIs(t, err, c.ExpectedErr)
 	} else {
+		expectedRefL2 := refToRune(refL2)
+		var expectedRefsL1 []rune
+		for _, ref := range nextRefL1s {
+			expectedRefsL1 = append(expectedRefsL1, refToRune(ref))
+		}
+
 		assert.NoError(t, err)
-		assert.Equal(t, expectedNextRefL1, nextRefL1, "expected %s (nr %d) but got %s (nr %d)", expectedNextRefL1.Hash[:1], expectedNextRefL1.Number, nextRefL1.Hash[:1], nextRefL1.Number)
-		assert.Equal(t, expectedRefL2, refL2, "expected %s (nr %d) but got %s (nr %d)", expectedRefL2.Hash[:1], expectedRefL2.Number, refL2.Hash[:1], refL2.Number)
+		assert.Equal(t, c.ExpectedNextRefsL1, string(expectedRefsL1), "Next L1 refs not equal")
+		assert.Equal(t, expectedRefL2, c.ExpectedRefL2, "Next L2 Head not equal")
 	}
 }
 
 func TestFindSyncStart(t *testing.T) {
 	testCases := []syncStartTestCase{
```

Comment:
If we make that `100` L1 blocks max configurable we can set it to something smaller for this test